File: src/main/java/io/poshidaev/toda/entity/GoalApproach.java
package io.poshidaev.toda.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;

@EqualsAndHashCode(callSuper = true)
@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class GoalApproach extends SingleIdEntity{

    @Temporal(value = TemporalType.DATE)
    private Date date;

    @ManyToOne(fetch = FetchType.LAZY)
    private Goal goal;
}

File: src/main/kotlin/io/poshidaev/toda/entity/GoalApproach.kt
package io.poshidaev.toda.entity

import java.util.*
import javax.persistence.*

@Entity
class GoalApproach : SingleIdEntity {
    constructor(id: Long?, status: GoalStatus, date: Date, goal: Goal?) : super(id) {
        this.status = status
        this.date = date
        this.goal = goal
    }

    constructor() : super()

    @Enumerated(EnumType.ORDINAL)
    var status: GoalStatus = GoalStatus.IN_PROGRESS

    @Temporal(value = TemporalType.DATE)
    var date: Date = Date()

    @ManyToOne(fetch = FetchType.LAZY)
    var goal: Goal? = null

}